apps/prairielearn/python/prairielearn.py

```diff
@@ -1628,12 +1629,20 @@ def is_correct_scalar_sf(a_sub, a_tru, digits=2):
     return (a_sub > lower_bound) & (a_sub < upper_bound)
 
 
-def get_uuid():
-    """get_uuid()
-
+def get_uuid() -> str:
+    """
     Returns the string representation of a new random UUID.
+    First character of this uuid is guaranteed to be an alpha
+    (at the expense of a slight loss in randomness).
+
+    This is done because certain web components need identifiers to
+    start with letters and not numbers.
     """
-    return str(uuid.uuid4())
+
+    uuid_string = str(uuid.uuid4())
+    random_char = random.choice("abcdef")
+
+    return random_char + uuid_string[1:]
```

Comment:
This loses a bit of randomness (not a big deal, but if you can make it, why not). How about:

```suggestion
    uuid_string = str(uuid.uuid4())

    # replaces 0-f range in first character with a-p range
    return chr(ord('a')+int(uuid_string[0], 16)) + uuid_string[1:]
```

Reply:
I deliberately chose to make the uuid this generated only use valid hex characters in case any client code depends on this property, at the expense of a loss in randomness. I think the loss is inconsequential, and I _really_ want to avoid breaking course code (not that I think this would break anything either way).